packages/ui/src/memberships/modals/BuyMembershipModal/BuyMembershipFormModal.tsx

```diff
@@ -152,6 +160,18 @@ export const BuyMembershipForm = ({
   ])
 
   const [validatorAccounts, setValidatorAccounts] = useState<Account[]>([])
+  const isValidValidatorAccount = useMemo(
```

Comment:
To use the `filter` it might be nice to expose the list of addresses:

```suggestion
  const validatorAddresses = useMemo(() => {
    if (!validatorAccountCandidate || !allValidatorsWithCtrlAcc || !allValidators) return

    return [...allValidatorsWithCtrlAcc, ...allValidators.map(({ address }) => address]
      .filter((element) => !!element) // Is this filter needed ?
      .map(encodeAddress)
  }, [allValidators, allValidatorsWithCtrlAcc, validatorAccountCandidate])

  const isValidValidatorAccount = useMemo(
```